### ISSUE TITLE
Is the code in place to check if a site passes its upgrade expiration date?

### ISSUE THREAD

**Issue by onaseef** [onaseef/yomobi#407]
Does the code already check for when a site passes its upgrade expiration date? So, for example, if a user pays for an annual subscription and then cancels it after 1 month, the expiration date is 11 months way. 

How does the code use the expiration date?  As in, does it check it every time the site is requested or is there a trigger that goes off on the date to make the site revert back?

When the date is passed, does the site go back to standard site?

**Comment by gilbert** [comment]
Right now there's no trigger to recalculate premium status. The logic to recalculate premium is already in place, it's just a matter of when to call it.

**Comment by onaseef** [comment]
So what happens when a site hits its upgrade expiration date, does it automatically revert back?

**Comment by gilbert** [comment]
Nope. I just found this [heroku scheduler add-on](https://devcenter.heroku.com/articles/scheduler), which will make it very easy to schedule this mass update one a day.

**Comment by onaseef** [comment]
So right now, once a site becomes premium, it stays that way forever? Why not put a check in place when a site is requested that happens if a site is premium. It would look at the expiration date and if it is less then DateTime.now, revert the site to standard?

**Comment by onaseef** [comment]
Using a schedule is also fine. I take it the schedule would run a program that checks a site upgrade expiration date and if it is less than now it sets the site-premium variable to false?

**Comment by gilbert** [comment]
Doing a check on every request would be expensive. You are right about the scheduler.

**Comment by gilbert** [comment]
There's now a rake task in place. When we push it to production, remind me to add it to the scheduler.

You can manually run it on your local machine like this:

```
$ rake recalculate_premiums
```

**Comment by onaseef** [comment]
Does this run in its own thread per tje Heroku recommendation so it doesn't freeze the UI?

**Comment by gilbert** [comment]
Yup, you can view the documentation in the link above.

**Comment by onaseef** [comment]
Excuse the repetition, but just to be certain, I'm reading this paragraph in the documentation:

Long-Running Jobs

Scheduled jobs are meant to execute short running tasks or enqueue longer running tasks into a background job queue. Anything that takes longer than a couple of minutes to complete should use a worker process to run.

Note that every scheduled job will execute irrespective of whether the previously scheduled job terminated or not - which may result in a lot of dyno hour usage.

This would:
1. run in the background job queue?
2. always terminate, so we don't have dynos going?

**Comment by gilbert** [comment]
It's a short task that always terminates. You can see for yourself by running it on heroku:

```
$ heroku run rake recalculate_premiums --app yomobi-test
```
